src/injectors/injector.ts

```diff
@@ -67,6 +68,48 @@ export abstract class InjectorBase implements Injector {
     }
 }
 
+function openInGitpod(e: MouseEvent | KeyboardEvent, inNewTab: boolean) {
+    const currentUrl = window.location.href;
+    window.open(`https://gitpod.io/#${currentUrl}`, inNewTab ? '_blank' : '_self');
+    e.preventDefault();
+    e.stopPropagation();
+}
+
+export async function rewritePeriodKeybindGitHub() {
+    const configProvider = await ConfigProvider.create();
+    const config = configProvider.getConfig();
+
+    if (config.rewritePeriodKeybind) {
+        document.querySelectorAll('.js-github-dev-shortcut, .js-github-dev-new-tab-shortcut').forEach((elem) => {
+            const new_element = elem.cloneNode(true) as HTMLElement;
+            elem.parentNode?.replaceChild(new_element, elem);
+            new_element.addEventListener('click', (e) => {
+                if (new_element && isVisible(new_element) && !confirm('Are you sure you want to open gitpod.io?')) {
+                    return;
+                }
+                openInGitpod(e, elem.classList.contains('js-github-dev-new-tab-shortcut') || config.openAsPopup);
+            });
+        });
+    }
+}
+
+export async function rewritePeriodKeybindGitLab() {
+    const configProvider = await ConfigProvider.create();
+    const config = configProvider.getConfig();
+
+    if (config.rewritePeriodKeybind) {
```

Comment:
Just in case in the future they stop using Mousetrap, we could check:
```suggestion
    if (config.rewritePeriodKeybind && "Mousetrap" in window) {
```

Reply:
Removing this suggestion because the "window.Mousetrap" doesn't exist in the extension context, [as explained by Filip in the discussion above.](https://github.com/gitpod-io/browser-extension/pull/54#discussion_r800524018)